generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/types/utils/functionReturnTypes.kt

```diff
@@ -38,12 +41,16 @@ import kotlin.reflect.KType
  *      Publisher<DataFetcherResult<T>>
  *      CompletableFuture<T>
  *      CompletableFuture<DataFetcherResult<T>>
+ *      List<DataFetcherResult<T>>
  */
 internal fun getWrappedReturnType(returnType: KType): KType {
     return when {
         returnType.isSubclassOf(DataFetcherResult::class) -> returnType.getTypeOfFirstArgument()
         returnType.isSubclassOf(Publisher::class) -> { checkTypeForDataFetcherResult(returnType) }
         returnType.isSubclassOf(CompletableFuture::class) -> { checkTypeForDataFetcherResult(returnType) }
+        returnType.isSubclassOf(List::class) && returnType.getTypeOfFirstArgument().isSubclassOf(DataFetcherResult::class) -> {
+            List::class.createType(listOf(KTypeProjection(KVariance.INVARIANT, getWrappedReturnType(returnType.getTypeOfFirstArgument()))))
```

Comment:
What happens if someone tries to return a `List< DataFetcherResult<List<String>>>`?

Reply:
I made the changes here to the unit test and everything looks good. Since the original author is not available, I will merge and then update the unit tests in a second PR